### PR TITLE
gh #47 L1 TV Settings issues by SoC Vendor

### LINKED ISSUE #47: L1 - TvSettings - Issue with new Vendor SoC
1) In Tc's test_l1_tvSettings_negative_SetCurrentComponentLuma,
VTS is expecting invalid value, but the current parameter passed to HAL API by VTS is valid, so VTS Testcase is getting failed.

Current code:
```
            if((tvDataComponentColor_t) componentColor !=(tvDataComponentColor_t) i)
            {
                bModeMatched = false;
                break;            
             }
```

Correct code
```

            if((tvDataComponentColor_t) componentColor !=(tvDataComponentColor_t) i)
            {
                bModeMatched = true;
                break;            
             }
```
             
Verify in all tests.

2) In test_l1_tvSettings_positive_GetMaxGainValue, and test_l1_tvSettings_negative_GetMaxGainValue
update logic to verify range
maxGain = GetMaxGainValue();  
UT_ASSERT_TRUE( (maxGain >= (2^10)) && ( maxGain <= (2^31)-1) );

### PR DESCRIPTION
**Pull request by akhilbhas**
* Range value updated
* corrected the valid value check

Closes #47

### CODE REVIEW

**Review by thanushreevani — APPROVED** [review]

**Review by Anbukannadhasan — COMMENTED** [review]
Please address the review questions

File: src/test_l1_tvSettings.c [hunk 1]
```diff
@@ -14239,7 +14238,7 @@ void test_l1_tvSettings_negative_SetCurrentComponentLuma (void)
             componentColor = (tvDataComponentColor_t)UT_KVP_PROFILE_GET_UINT32(keyValue);
             if((tvDataComponentColor_t) componentColor !=(tvDataComponentColor_t) i)
             {
-                bModeMatched = false;
+                bModeMatched = true;
                 break;
```
[the remark this answers is not in the document]
Comment (akhilbhas) [reply src/test_l1_tvSettings.c:14243]: I those cases, we are curre4ntly reading the value twice and comparing it.

File: src/test_l1_tvSettings.c [hunk 2]
```diff
@@ -14239,7 +14238,7 @@ void test_l1_tvSettings_negative_SetCurrentComponentLuma (void)
             componentColor = (tvDataComponentColor_t)UT_KVP_PROFILE_GET_UINT32(keyValue);
             if((tvDataComponentColor_t) componentColor !=(tvDataComponentColor_t) i)
             {
-                bModeMatched = false;
+                bModeMatched = true;
```
[the remark this answers is not in the document]
Comment (akhilbhas) [reply src/test_l1_tvSettings.c:14241]: in this case, actual issue was the variable name passed to the getAPI , We passed same variable twice and comparing with two different variables in comparison time.
I have updated code to pass proper variable in two calls (ggain and ggainRetry) respectively.

**Review by Anbukannadhasan — APPROVED** [review]
Approved

**Review by srinivasgtl — APPROVED** [review]
shipit

**Pull request merged**